lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2799,6 +2800,9 @@ class HSAContext final : public KalmarContext
     */
     hsa_agent_t host;
 
+    cxlProfiler profiler;
+    cxlMarker contextMarker = CXL_MARKER_CLASS;
```

Comment:
@scchan If I understand you correctly, it seems:
- `hc_profile` is meant to be a proxy to encapsulate all interactions with CodeXL,
- and `hc_profile` can be used to provide `hc_am` and `mcwamp_hsa` themselves, plus HCC applications.

If that's the case I think it is fine to build `hc_profile` as s standalone library, *but* we should:
- make this a CMake option to be switched on/off at configure time
- in `hcc-config`, turn on/off `-lhc_profile` inside `--ldflags` based on the CMake option
- guard the dependency to `hc_profile` inside `hc_am` and `mcwamp_hsa` based on the CMake option
- in `hcc-config`, perhaps add another flag `--profile` to explicitly specify `-lhc_profile` for user applications